### PR TITLE
fix(bindings): Repair Releases

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Fixes golang codegen to dup static file output to rust crate so it can be included in the crate upload.

Also cleans up crates and their dependencies quite a bit - removes a lot of cruft and splits code out nicely into different modules.

The ci release workflow should now work, will put that up in another pr.

### CODE REVIEW

**Review by clabby — APPROVED** [review]
Rust code lgtm, pending go reviewer's approval.

File: bindings/rust-bindings/etc/chainList.toml
Comment (bitwiseguy) [inline bindings/rust-bindings/etc/chainList.toml:1]: Can we just use the existing `<repo-root>/chainList.toml` file instead of duplicating here?

File: superchain/configs/configs.toml
Comment (bitwiseguy) [inline superchain/configs/configs.toml:5]: This file is only currently used by the rust code. Could we remove this entire file (and the codegen code that creates it) if you prefer for the file to be written to the rust dir? Or do you think this file will be useful in the future for other languages to create their own bindings?

**Review by bitwiseguy — APPROVED** [review]

File: bindings/rust-bindings/etc/chainList.toml
Comment (refcell) [reply bindings/rust-bindings/etc/chainList.toml:1]: No we can't unfortunately because then it won't be included in the crates release and when someone uses the crate, it won't build because the file is missing. I will continue trying to figure out a work around but this is the best solution I found after a few hours

File: superchain/configs/configs.toml
Comment (refcell) [reply superchain/configs/configs.toml:5]: Ah that's a great idea actually, will do.

File: bindings/rust-bindings/etc/chainList.toml
Comment (bitwiseguy) [reply bindings/rust-bindings/etc/chainList.toml:1]: Ok no worries. Doesn't need to block this pr, just a potential future cleanup

**Pull request merged**
